lib/ui/login/login_screen.dart

```diff
@@ -108,102 +122,129 @@ class _LoginScreenState extends State<LoginScreen>
-                Padding(
-                  padding: const EdgeInsets.only(
-                    left: AppDimensions.spacing24,
-                    right: AppDimensions.spacing24,
-                    bottom: AppDimensions.spacing20,
-                  ),
-                  child: _configuredTextField(context, false),
+                child: _configuredTextField(context, true),
+              ),
+              Padding(
+                padding: const EdgeInsets.only(
+                  left: AppDimensions.spacing24,
+                  right: AppDimensions.spacing24,
+                  bottom: AppDimensions.spacing20,
+                ),
+                child: _configuredTextField(context, false),
+              ),
+              Padding(
+                padding: const EdgeInsets.symmetric(
+                  horizontal: AppDimensions.spacing24,
                 ),
-                Padding(
-                  padding: const EdgeInsets.symmetric(
-                    horizontal: AppDimensions.spacing24,
-                  ),
-                  child: SizedBox(
-                    height: 56.0,
-                    width: double.infinity,
-                    child: _loginButton(context),
-                  ),
+                child: SizedBox(
+                  height: 56.0,
+                  width: double.infinity,
+                  child: _loginButton(context),
                 ),
-              ],
-            ),
+              ),
+            ],
           ),
-        ],
-      );
+        ),
+      ],
+    );
+  }
 
   @override
   Widget build(BuildContext context) {
+    final isLoginSucceed = ref.watch(_loginStatusProvider).value ?? false;
+    final errorMessage = ref.watch(_loginErrorStatusProvider).value ?? "";
     return Material(
-      child: _buildLoginScreen(context),
+      child: Scaffold(
+        body: ref.watch(loginViewModelProvider).when(
+              init: () =>
+                  _buildLoginScreen(context, isLoginSucceed, errorMessage),
+              loading: () =>
+                  _buildLoginScreen(context, isLoginSucceed, errorMessage),
+              loginSuccess: () =>
+                  _buildLoginScreen(context, isLoginSucceed, errorMessage),
+              loginError: () =>
+                  _buildLoginScreen(context, isLoginSucceed, errorMessage),
+            ),
+      ),
     );
   }
 
-  // TODO: remove this dummy method in [#10]
   void logIn() async {
-    final LoginInput input = LoginInput(
-      email: _emailTextFieldController.text,
-      password: _passwordTextFieldController.text,
-    );
-    final result = await _loginUseCase.call(input);
-    if (result is Success<LoginModel>) {
-      // final loginData = result.value;
-      // print("LOGIN SUCCESS DATA\n$loginData");
-      // save login data
-      // Move to survey page
-    } else {
-      // show login error message
+    if (_isLoading) {
+      return;
     }
+    setState(() {
+      _isLoading = true;
+    });
```

Comment:
The loading should be controlled by the ViewModel with the LoginState. We should not use a local `_isLoading` for that 🙏